### PR TITLE
[clang]MveEmitter:Pass Args as const references *bad rebase, please ignore*

### LINKED ISSUE #89192: clang/utils/TableGen/MveEmitter.cpp: 2 * pointless copy ?
Static analyser cppcheck says:

1.

clang/utils/TableGen/MveEmitter.cpp:663:51: performance: Function parameter 'IntegerArgs' should be passed by const reference. [passedByValue]

Source code is

  IRBuilderResult(StringRef CallPrefix, std::vector<Ptr> Args,
                  std::set<unsigned> AddressArgs,
                  std::map<unsigned, std::string> IntegerArgs)

2.

clang/utils/TableGen/MveEmitter.cpp:731:38: performance: Function parameter 'Args' should be passed by const reference. [passedByValue]

Source code is

 IRIntrinsicResult(StringRef IntrinsicID, std::vector<const Type *> ParamTypes,
                    std::vector<Ptr> Args)

### PR DESCRIPTION
**Pull request by aniplcc**
Closes  #89192

### CODE REVIEW

**Comment by aniplcc** [comment]
cc @RKSimon

File: clang/utils/TableGen/MveEmitter.cpp [hunk 1]
```diff
@@ -660,7 +660,7 @@ class IRBuilderResult : public Result {
   std::map<unsigned, std::string> IntegerArgs;
   IRBuilderResult(StringRef CallPrefix, std::vector<Ptr> Args,
```
Comment (RKSimon) [inline clang/utils/TableGen/MveEmitter.cpp:661]: std::vector<Ptr> Args?

File: clang/utils/TableGen/MveEmitter.cpp [hunk 2]
```diff
@@ -660,7 +660,7 @@ class IRBuilderResult : public Result {
...
                   std::set<unsigned> AddressArgs,
```
Comment (RKSimon) [inline clang/utils/TableGen/MveEmitter.cpp:662]: std::set<unsigned> AddressArgs?

File: clang/utils/TableGen/MveEmitter.cpp [hunk 3]
```diff
@@ -728,7 +728,7 @@ class IRIntrinsicResult : public Result {
   std::vector<const Type *> ParamTypes;
   std::vector<Ptr> Args;
   IRIntrinsicResult(StringRef IntrinsicID, std::vector<const Type *> ParamTypes,
```
Comment (RKSimon) [inline clang/utils/TableGen/MveEmitter.cpp:730]: std::vector<const Type *> ParamTypes?

**Pull request closed without merging**